src/lib.rs

```diff
@@ -393,6 +396,29 @@ pub enum _CastType {
 }
 
 impl DstLayout {
+    /// The maximum theoretic possible alignment of a type.
+    ///
+    /// For compatibility with future Rust versions, this is defined as the
+    /// maximum power-of-two that fits into a `usize`. See also
+    /// [`DstLayout::CURRENT_MAX_ALIGN`].
+    const THEORETICAL_MAX_ALIGN: NonZeroUsize =
+        match NonZeroUsize::new(1 << (POINTER_WIDTH_BITS - 1)) {
```

Comment:
This can be expressed more simply as:
```suggestion
        match NonZeroUsize::new(isize::MIN as usize) {
```
...which would let us avoid introducing the `POINTER_WIDTH_BITS` constant. 

Is this too magical, @joshlf?

Reply:
If you're hoping to avoid the constant, maybe just inline its definition? It'd accomplish that goal while keeping this pretty self-explanatory.

Reply:
I did that initially, but it looked a little too messy to my eyes.

Reply:
Up to you, but IMO `isize::MIN as usize` is too magical.